app/controllers/api/base_controller.rb

```diff
@@ -1,3 +1,14 @@
 class Api::BaseController < ApplicationController
   layout nil
+
+  # Method to log login attempts
+  def log_login_attempt(user_id:, attempted_at:, successful:, ip_address:)
```

Comment:
Consider adding error handling for the 'log_login_attempt' method to manage cases where logging fails, ensuring that the system remains robust and that failures are properly reported or handled.